### PR TITLE
DATAUP-330 fixed narrative viewers tests and codebase

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

This was, and still is, kinda ugly.

The core problem here was that the NarrativeViewers module would, as soon as it was loaded, make some calls to the Narrative Method Store, regardless of whether that was needed or not. So the main fix here was to make it only do that when the `createViewer` function is called. And then cache the result for 5 minutes.

A secondary problem was actually finding a code path that would run it for real so it could be tested live. Turns out this is only invoked by the decrepit `kbaseNarrativeDataCell` widget, which has since been supplanted by other mechanisms. The only narrative I could find to invoke that was an outdated one that was automatically updated to other mechanisms, yet still uses that module. It's here: https://ci.kbase.us/narrative/630 , shared with @ialarmedalien and @eapearson. It's the first cell that triggers it.

I'd like to completely remove that hunk of code, but since it's built in to some legacy narratives, I don't think we can yet. And I gotta move on to other things this sprint, so band-aid it is. I tried to keep changes to a minimum, other than some minor syntax updates (var -> const / let, arrow functions as appropriate).

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-330
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
Try out the linked narrative, and/or run the unit tests.
- [x] Tests pass locally and in GitHub Actions

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

### CODE REVIEW

File: test/unit/spec/narrative_core/narrativeViewersSpec.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            expect(Viewers).toBeDefined();
+
+            ['getViewerInfo', 'createViewer', 'defaultViewer'].forEach(fn => {
+                expect(Viewers[fn]).toBeDefined();
+            });
+        });
+
+        it('should load viewer info as a promise', () => {
+            return Viewers.getViewerInfo().then((info) => {
+                expect(info.viewers).toBeDefined();
```
Comment (ialarmedalien) [inline test/unit/spec/narrative_core/narrativeViewersSpec.js:103]: You could put these in an array and then iterate through -- saves a line and makes it easy to amend.

File: test/unit/spec/narrative_core/narrativeViewersSpec.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            'Module.Type2': {
+                view_method_ids: ['Viewers/missing_viewer_app'],
+                landing_page_url_prefix: 'type2',
+                name: 'Type 2'
+            },
+            'Module.Type3': {
+                view_method_ids: ['Viewers/error_viewer_app'],
+                landing_page_url_prefix: 'type3',
+                name: 'Type 3'
+            }
```
Comment (ialarmedalien) [inline test/unit/spec/narrative_core/narrativeViewersSpec.js:38]: is this data used/tested anywhere?

Comment (briehl) [reply test/unit/spec/narrative_core/narrativeViewersSpec.js:38]: Yes, that's the mocked result from NMS.list_categories (the mocked out type information)

File: test/unit/spec/narrative_core/narrativeViewersSpec.js (hunk 1 above)
Comment (briehl) [reply test/unit/spec/narrative_core/narrativeViewersSpec.js:103]: Ah, sure, I didn't see that. I didn't touch the older tests, just migrated to a different spot. I'll update.

**Review by ialarmedalien — APPROVED** [review]
🦖 👍

**Pull request merged**